File: SimpleAccounting/app/src/main/java/com/emmanuelmess/simpleaccounting/activities/DonateActivity.java
package com.emmanuelmess.simpleaccounting.activities;

import android.content.ActivityNotFoundException;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.support.design.widget.Snackbar;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListAdapter;
import android.widget.ListView;
import android.widget.SimpleAdapter;

import com.emmanuelmess.simpleaccounting.R;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

/**
 * @author Emmanuel
 *         on 12/5/2017, at 23:39.
 */

public class DonateActivity extends AppCompatActivity {

	private static final String BITCOIN_DIRECTION = "1HFhPxH9bqMKvs44nHqXjEEPC2m7z1V8tW";
	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_donate);

		ActionBar actionBar = getSupportActionBar();
		if (actionBar != null) {
			// Show the Up button in the action bar.
			actionBar.setDisplayHomeAsUpEnabled(true);
		}

		List<HashMap<String, String>> m = new ArrayList<>();
		HashMap<String, String> h = new HashMap<>();
		h.put("title", getString(R.string.bitcoin));
		h.put("summary", BITCOIN_DIRECTION);
		m.add(h);
		ListAdapter adapter = new SimpleAdapter(this, m, R.layout.item_donate,
				new String[]{"title", "summary"}, new int[]{R.id.title, R.id.summary});

		// create a new ListView, set the adapter and item click listener
		ListView listViewItems = (ListView) findViewById(R.id.listView);
		listViewItems.setAdapter(adapter);
		listViewItems.setOnItemClickListener(new AdapterView.OnItemClickListener() {
			@Override
			public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
				switch (position) {
					case 0:
						try {
							String url = "bitcoin:" + BITCOIN_DIRECTION + "?amount=0.0005";
							Intent i = new Intent(Intent.ACTION_VIEW);
							i.setData(Uri.parse(url));
							startActivity(i);
						} catch (ActivityNotFoundException e) {
							Snackbar.make(listViewItems, R.string.no_bitcoin_app,
									Snackbar.LENGTH_LONG).show();
						}
						break;
				}
			}
		});

	}
}

File: SimpleAccounting/app/src/main/java/com/emmanuelmess/simpleaccounting/activities/DonateActivity.kt
package com.emmanuelmess.simpleaccounting.activities

import android.content.ActivityNotFoundException
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.emmanuelmess.simpleaccounting.R
import com.emmanuelmess.simpleaccounting.patreon.PatreonController
import com.google.android.material.snackbar.Snackbar

class DonateActivity : AppCompatActivity() {

	companion object {
		private const val BITCOIN_DIRECTION = "1HFhPxH9bqMKvs44nHqXjEEPC2m7z1V8tW"
	}

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(R.layout.activity_donate)

		supportActionBar?.setDisplayHomeAsUpEnabled(true)
	}

	fun onClickDonate(view: View) {
		PatreonController.openPage(this)
	}

	fun onClickBitcoin(view: View) {
		try {
			with(Intent(Intent.ACTION_VIEW)) {
				data = Uri.parse("bitcoin:$BITCOIN_DIRECTION?amount=0.0005")
				startActivity(this@with)
			}
		} catch (e: ActivityNotFoundException) {
			Snackbar.make(view, R.string.no_bitcoin_app, Snackbar.LENGTH_LONG).show()
		}
	}

	fun onClickPaypal(view: View) {
		Intent(Intent.ACTION_VIEW).let {
			it.data = Uri.parse("https://www.paypal.com/invoice/p/#NNT9XYPXLB69BG3A")
			startActivity(it)
		}
	}

	fun onClickKofi(view: View) {
		Intent(Intent.ACTION_VIEW).let {
			it.data = Uri.parse("https://ko-fi.com/emmanuelmess")
			startActivity(it)
		}
	}

}